### ISSUE TITLE
Metric: Available Dataset and Code Score (dataset_and_code_score)

### ISSUE THREAD

**Issue by vdudhaiy** [ECE461ProjTeam/ModelReuseCLI#19]
We will use the HuggingFace API to fetch information about the datasets (if available) and the code in the model repository. At the same time, we will use an LLM to go through the files and look for training or testing scripts, a README file, requirements.txt. We will also use the LLM to go through the README and determine whether the information is well documented. 

If the model has information about its datasets, it will gain 0.25 points. The dataset information is crucial in the future evaluation of the model.
If the model has a requirements.txt, it will gain 0.10 points. This file plays an important role in reproducibility.  
If the model has relevant training files, it will gain 0.25 points. 
If the model has relevant testing files, it will gain 0.15 points. Training files are integral for understanding, replication, and finetuning while the testing files are, while still important, built upon the structure of the training files. 
If the README file exists and is well documented, it will gain another 0.25 points. Partial credit is allowed.

If a model fails all the conditions, it will receive a score of 0.

**Comment by vdudhaiy** [comment]
From the project document:
- Inspect config.json, model_index.json, or README.md for architecture, 
training details, datasets used.
- Check the size of weights (pytorch_model.bin, tf_model.h5, etc.) for 
deployability.
- Verify presence of tokenizer/vocabulary files.
- Static analysis → Parse config.json / metadata for structure.

So maybe it does make sense to use isomorphic-git for this metric as well?
